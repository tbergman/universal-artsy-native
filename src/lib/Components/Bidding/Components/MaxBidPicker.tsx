import React from "react"
import { Picker, PickerProperties } from "react-native"
import styled from "styled-components/native"

import { Bid } from "../types"

export interface MaxBidPickerProps extends PickerProperties {
  bids: ReadonlyArray<Bid>
}

export class MaxBidPicker extends React.Component<MaxBidPickerProps> {
  render() {
    return (
      <StyledPicker {...this.props} onValueChange={this.props.onValueChange} selectedValue={this.props.selectedValue}>
        {this.props.bids.map((bid, index) => <Picker.Item key={index} value={index} label={bid.display} />)}
      </StyledPicker>
    )
  }
}

const StyledPicker = styled.Picker`
  width: 100%;
`
