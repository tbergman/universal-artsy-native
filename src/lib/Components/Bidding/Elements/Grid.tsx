import React from "react"
import { Flex } from "../Elements/Flex"

export const Row = props => <Flex flexDirection="row" justifyContent="space-between" alignItems="center" {...props} />
export const Col = props => <Flex flex={1} {...props} />
