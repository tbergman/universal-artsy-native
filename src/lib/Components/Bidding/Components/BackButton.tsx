import React from "react"
import { NavigatorIOS, TouchableWithoutFeedback, ViewProperties } from "react-native"

import { Image } from "../Elements/Image"
import { theme } from "../Elements/Theme"

interface ContainerWithBackButtonProps extends ViewProperties {
  navigator: NavigatorIOS
}

export class BackButton extends React.Component<ContainerWithBackButtonProps> {
  goBack() {
    this.props.navigator.pop()
  }

  render() {
    return (
      <TouchableWithoutFeedback onPress={() => this.goBack()}>
        <Image
          position="absolute"
          top={theme.space[3]}
          left={theme.space[3]}
          source={require("../../../../../images/angle-left.png")}
          style={{ zIndex: 10 }} // Here the style prop is intentionally used to avoid making zIndex too handy.
        />
      </TouchableWithoutFeedback>
    )
  }
}
