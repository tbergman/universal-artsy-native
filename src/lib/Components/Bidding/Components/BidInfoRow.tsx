import React from "react"
import { TouchableWithoutFeedback } from "react-native"

import { FlexProps } from "../Elements/Flex"
import { Col, Row } from "../Elements/Grid"
import { Sans12, Serif16, SerifSemibold16 } from "../Elements/Typography"

interface BidInfoRowProps extends FlexProps {
  label: string
  value: string
  onPress?: () => void
}

export class BidInfoRow extends React.Component<BidInfoRowProps> {
  render() {
    const { label, value, onPress, ...props } = this.props

    return (
      <TouchableWithoutFeedback onPress={onPress}>
        <Row p={4} pb={3} mb={1} {...props}>
          <Col>
            <SerifSemibold16>{label}</SerifSemibold16>
          </Col>

          <Col alignItems="flex-end">{value && <Serif16 numberOfLines={1}>{value}</Serif16>}</Col>

          <Col alignItems="flex-end" flexGrow={0} flexShrink={0} flexBasis="auto" flex={null}>
            <Sans12 color="purple100" ml={3} mb={1}>
              {Boolean(value) ? "Edit" : "Add"}
            </Sans12>
          </Col>
        </Row>
      </TouchableWithoutFeedback>
    )
  }
}
