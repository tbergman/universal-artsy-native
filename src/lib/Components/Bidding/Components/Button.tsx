import React from "react"
import { StyleSheet } from "react-native"

import PrimaryBlack, { PrimaryBlackProps } from "lib/Components/Buttons/PrimaryBlack"
import { ButtonProps, SecondaryOutlineButton } from "../../Buttons"
import { Flex } from "../Elements/Flex"

export class Button extends React.Component<PrimaryBlackProps> {
  render() {
    const { textStyle, ...props } = this.props

    return (
      <Flex height={50}>
        <PrimaryBlack textStyle={[textStyle]} {...props} />
      </Flex>
    )
  }
}

export class BidGhostButton extends React.Component<ButtonProps> {
  render() {
    const { style, textStyle, ...props } = this.props

    return (
      <SecondaryOutlineButton
        style={[styles.default, styles.secondaryOutlineStyles, style]}
        textStyle={[textStyle]}
        {...props}
      />
    )
  }
}

const styles = StyleSheet.create({
  default: {
    height: 50,
  },
  secondaryOutlineStyles: {
    borderRadius: 2,
    borderWidth: 2,
  },
})
