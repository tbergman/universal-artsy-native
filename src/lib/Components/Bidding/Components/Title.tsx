import React from "react"

import { SerifSemibold22t } from "../Elements/Typography"

export const Title = props => <SerifSemibold22t m={4} textAlign="center" {...props} />
