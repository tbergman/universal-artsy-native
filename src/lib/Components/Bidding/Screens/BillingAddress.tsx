import React from "react"

import { Schema, screenTrack, track } from "../../../utils/track"

import { Dimensions, KeyboardAvoidingView, NavigatorIOS, ScrollView, TouchableWithoutFeedback } from "react-native"

import { Flex } from "../Elements/Flex"
import { Sans12, Serif16 } from "../Elements/Typography"

import { validatePresence } from "../Validators"

import { BackButton } from "../Components/BackButton"
import { BiddingThemeProvider } from "../Components/BiddingThemeProvider"
import { Button } from "../Components/Button"
import { Container } from "../Components/Containers"
import { Input } from "../Components/Input"
import { Title } from "../Components/Title"
import { Address, Country } from "../types"

import { SelectCountry } from "./SelectCountry"

interface StyledInputInterface {
  /** The object which styled components wraps */
  root?: {
    focus?: () => void
    blur?: () => void
  }
}

const StyledInput = ({ label, error, ...props }) => (
  <Flex mb={4}>
    <Serif16 mb={2}>{label}</Serif16>
    <Input mb={3} error={Boolean(error)} {...props} />
    {error && <Sans12 color="red100">{error}</Sans12>}
  </Flex>
)

interface BillingAddressProps {
  onSubmit?: (values: Address) => void
  navigator?: NavigatorIOS
  billingAddress?: Address
}

interface BillingAddressState {
  values: Address
  errors: {
    fullName?: string
    addressLine1?: string
    addressLine2?: string
    city?: string
    state?: string
    country?: string
    postalCode?: string
  }
}

@screenTrack({
  context_screen: Schema.PageNames.BidFlowBillingAddressPage,
  context_screen_owner_type: null,
})
export class BillingAddress extends React.Component<BillingAddressProps, BillingAddressState> {
  private addressLine1: StyledInputInterface
  private addressLine2: StyledInputInterface
  private city: StyledInputInterface
  private stateProvinceRegion: StyledInputInterface
  private postalCode: StyledInputInterface

  constructor(props) {
    super(props)

    this.state = {
      values: { ...this.props.billingAddress },
      errors: {},
    }
  }

  validateAddress(address: Address) {
    const { fullName, addressLine1, city, state, country, postalCode } = address

    return {
      fullName: validatePresence(fullName),
      addressLine1: validatePresence(addressLine1),
      city: validatePresence(city),
      state: validatePresence(state),
      country: validatePresence(country && country.shortName),
      postalCode: validatePresence(postalCode),
    }
  }

  validateField(field: string) {
    this.setState({
      errors: { ...this.state.errors, [field]: this.validateAddress(this.state.values)[field] },
    })
  }

  onSubmit() {
    const errors = this.validateAddress(this.state.values)

    if (Object.keys(errors).filter(key => errors[key]).length > 0) {
      this.setState({ errors })
    } else {
      this.submitValidAddress()
    }
  }

  onCountrySelected(country: Country) {
    const values = { ...this.state.values, country }

    this.setState({
      values,
      errors: {
        ...this.state.errors,
        country: this.validateAddress(values).country,
      },
    })
  }

  @track({
    action_type: Schema.ActionTypes.Success,
    action_name: Schema.ActionNames.BidFlowSaveBillingAddress,
  })
  submitValidAddress() {
    this.props.onSubmit(this.state.values)
    this.props.navigator.pop()
  }

  presentSelectCountry() {
    this.props.navigator.push({
      component: SelectCountry,
      title: "",
      passProps: {
        country: this.state.values.country,
        onCountrySelected: this.onCountrySelected.bind(this),
      },
    })
  }

  render() {
    const errorForCountry = this.state.errors.country

    // TODO: Remove this once React Native has been updated
    const isPhoneX = Dimensions.get("window").height === 812 && Dimensions.get("window").width === 375
    const defaultVerticalOffset = isPhoneX ? 30 : 15

    return (
      <BiddingThemeProvider>
        <KeyboardAvoidingView behavior="padding" keyboardVerticalOffset={defaultVerticalOffset} style={{ flex: 1 }}>
          <BackButton navigator={this.props.navigator} />

          <ScrollView>
            <Container>
              <Title mt={0} mb={6}>
                Your billing address
              </Title>

              <StyledInput
                {...this.defaultPropsForInput("fullName")}
                label="Full name"
                placeholder="Add your full name"
                autoFocus={true}
                onSubmitEditing={() => this.addressLine1.root.focus()}
              />

              <StyledInput
                {...this.defaultPropsForInput("addressLine1")}
                label="Address line 1"
                placeholder="Add your street address"
                onSubmitEditing={() => this.addressLine2.root.focus()}
              />

              <StyledInput
                {...this.defaultPropsForInput("addressLine2")}
                label="Address line 2 (optional)"
                placeholder="Add your apt, floor, suite, etc."
                onSubmitEditing={() => this.city.root.focus()}
              />

              <StyledInput
                {...this.defaultPropsForInput("city")}
                label="City"
                placeholder="Add your city"
                onSubmitEditing={() => this.stateProvinceRegion.root.focus()}
              />

              <StyledInput
                {...this.defaultPropsForInput("state")}
                label="State, Province, or Region"
                placeholder="Add state, province, or region"
                onSubmitEditing={() => this.postalCode.root.focus()}
                inputRef={el => (this.stateProvinceRegion = el)}
              />

              <StyledInput
                {...this.defaultPropsForInput("postalCode")}
                label="Postal code"
                placeholder="Add your postal code"
                onSubmitEditing={() => this.presentSelectCountry()}
              />

              <Flex mb={4}>
                <Serif16 mb={2}>Country</Serif16>

                <TouchableWithoutFeedback onPress={() => this.presentSelectCountry()}>
                  <Flex mb={3} p={3} pb={2} border={1} borderColor={errorForCountry ? "red100" : "black10"}>
                    {this.state.values.country ? (
                      <Serif16>{this.state.values.country.longName}</Serif16>
                    ) : (
                      <Serif16 color="black30">Select your country</Serif16>
                    )}
                  </Flex>
                </TouchableWithoutFeedback>

                {errorForCountry && <Sans12 color="red100">{errorForCountry}</Sans12>}
              </Flex>

              <Button text="Add billing address" onPress={() => this.onSubmit()} />
            </Container>
          </ScrollView>
        </KeyboardAvoidingView>
      </BiddingThemeProvider>
    )
  }

  private defaultPropsForInput(field: string) {
    return {
      autoCapitalize: "words",
      error: this.state.errors[field],
      inputRef: el => (this[field] = el),
      onBlur: () => this.validateField(field),
      onChangeText: value => this.setState({ values: { ...this.state.values, [field]: value } }),
      returnKeyType: "next",
      value: this.state.values[field],
    }
  }
}
