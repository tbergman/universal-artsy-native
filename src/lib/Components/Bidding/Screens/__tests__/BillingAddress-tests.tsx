import React from "react"
import { TextInput, TouchableWithoutFeedback } from "react-native"
import * as renderer from "react-test-renderer"

import { Button } from "../../Components/Button"
import { Sans12, Serif16 } from "../../Elements/Typography"
import { BillingAddress } from "../BillingAddress"

import { FakeNavigator } from "../../__tests__/Helpers/FakeNavigator"

const selectCountry = (component, navigator, country) => {
  // The second `<TouchableWithoutFeedback>` is a button that pushes a new `<SelectCountry>` instance.
  component.root.findAllByType(TouchableWithoutFeedback)[1].instance.props.onPress()

  navigator.nextStep().root.instance.props.onCountrySelected(country)
}

it("renders properly", () => {
  const component = renderer.create(<BillingAddress />).toJSON()
  expect(component).toMatchSnapshot()
})

it("shows an error message for each field", () => {
  const component = renderer.create(<BillingAddress />)

  component.root.findByType(Button).instance.props.onPress()

  expect(errorTextComponent(component, "Full name").props.children).toEqual("This field is required")
  expect(errorTextComponent(component, "Address line 1").props.children).toEqual("This field is required")
  expect(errorTextComponent(component, "City").props.children).toEqual("This field is required")
  expect(errorTextComponent(component, "State, Province, or Region").props.children).toEqual("This field is required")
  expect(errorTextComponent(component, "Postal code").props.children).toEqual("This field is required")
})

it("calls the onSubmit() callback with billing address when ADD BILLING ADDRESS is tapped", () => {
  const fakeNavigator = new FakeNavigator()
  const onSubmitMock = jest.fn()

  const component = renderer.create(<BillingAddress onSubmit={onSubmitMock} navigator={fakeNavigator as any} />)

  textInputComponent(component, "Full name").props.onChangeText("Yuki Stockmeier")
  textInputComponent(component, "Address line 1").props.onChangeText("401 Broadway")
  textInputComponent(component, "Address line 2 (optional)").props.onChangeText("25th floor")
  textInputComponent(component, "City").props.onChangeText("New York")
  textInputComponent(component, "State, Province, or Region").props.onChangeText("NY")
  textInputComponent(component, "Postal code").props.onChangeText("10013")
  selectCountry(component, fakeNavigator, billingAddress.country)

  component.root.findByType(Button).instance.props.onPress()

  expect(onSubmitMock).toHaveBeenCalledWith(billingAddress)
})

it("updates the validation for country when coming back from the select country screen", () => {
  const fakeNavigator = new FakeNavigator()

  const component = renderer.create(<BillingAddress onSubmit={() => null} navigator={fakeNavigator as any} />)

  textInputComponent(component, "Full name").props.onChangeText("Yuki Stockmeier")
  textInputComponent(component, "Address line 1").props.onChangeText("401 Broadway")
  textInputComponent(component, "Address line 2 (optional)").props.onChangeText("25th floor")
  textInputComponent(component, "City").props.onChangeText("New York")
  textInputComponent(component, "State, Province, or Region").props.onChangeText("NY")
  textInputComponent(component, "Postal code").props.onChangeText("10013")

  component.root.findByType(Button).instance.props.onPress()

  expect(component.root.findByType(Sans12).props.children).toEqual("This field is required")

  selectCountry(component, fakeNavigator, billingAddress.country)

  // The <Sans12> instances in the BillingAddress screen display error messages
  expect(component.root.findAllByType(Sans12).length).toEqual(0)
})

it("pre-fills the fields if initial billing address is provided", () => {
  const component = renderer.create(<BillingAddress billingAddress={billingAddress} />)

  expect(textInputComponent(component, "Full name").props.value).toEqual("Yuki Stockmeier")
  expect(textInputComponent(component, "Address line 1").props.value).toEqual("401 Broadway")
  expect(textInputComponent(component, "Address line 2 (optional)").props.value).toEqual("25th floor")
  expect(textInputComponent(component, "City").props.value).toEqual("New York")
  expect(textInputComponent(component, "State, Province, or Region").props.value).toEqual("NY")
  expect(textInputComponent(component, "Postal code").props.value).toEqual("10013")

  const countryField = component.root.findAllByType(Serif16)[7]

  expect(countryField.props.children).toEqual("United States")
})

const errorTextComponent = (component, label) => findFieldForInput(component, { label }).findByType(Sans12)

const textInputComponent = (component, label) => findFieldForInput(component, { label }).findByType(TextInput)

const findFieldForInput = (component, { label }) => component.root.findByProps({ label })

const billingAddress = {
  fullName: "Yuki Stockmeier",
  addressLine1: "401 Broadway",
  addressLine2: "25th floor",
  city: "New York",
  state: "NY",
  postalCode: "10013",
  country: {
    longName: "United States",
    shortName: "US",
  },
}
