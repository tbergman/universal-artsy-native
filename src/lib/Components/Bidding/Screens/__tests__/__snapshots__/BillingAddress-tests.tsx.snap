// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<View
  enabled={true}
  keyboardVerticalOffset={15}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Image
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    left={10}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    position="absolute"
    source={
      Object {
        "testUri": "../../../images/angle-left.png",
      }
    }
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "zIndex": 10,
        },
      ]
    }
    testID={undefined}
    top={10}
  />
  <RCTScrollView>
    <View>
      <View
        flex={1}
        flexDirection="column"
        justifyContent="space-between"
        m={4}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          fontSize={5}
          lineHeight={7}
          m={4}
          mb={6}
          mt={0}
          style={
            Array [
              Object {
                "fontFamily": "AGaramondPro-Semibold",
                "fontSize": 22,
                "lineHeight": 28,
                "marginBottom": 40,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 0,
                "textAlign": "center",
              },
              undefined,
            ]
          }
          textAlign="center"
        >
          Your billing address
        </Text>
        <View
          mb={4}
          style={
            Array [
              Object {
                "marginBottom": 20,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={3}
            lineHeight={5}
            mb={2}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 5,
                },
                undefined,
              ]
            }
          >
            Full name
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            autoFocus={true}
            border={1}
            borderColor="black10"
            error={false}
            fontSize={3}
            inputRef={[Function]}
            mb={3}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            p={3}
            pb={2}
            placeholder="Add your full name"
            returnKeyType="next"
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingBottom": 5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ]
            }
            value={undefined}
          />
        </View>
        <View
          mb={4}
          style={
            Array [
              Object {
                "marginBottom": 20,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={3}
            lineHeight={5}
            mb={2}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 5,
                },
                undefined,
              ]
            }
          >
            Address line 1
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            border={1}
            borderColor="black10"
            error={false}
            fontSize={3}
            inputRef={[Function]}
            mb={3}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            p={3}
            pb={2}
            placeholder="Add your street address"
            returnKeyType="next"
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingBottom": 5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ]
            }
            value={undefined}
          />
        </View>
        <View
          mb={4}
          style={
            Array [
              Object {
                "marginBottom": 20,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={3}
            lineHeight={5}
            mb={2}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 5,
                },
                undefined,
              ]
            }
          >
            Address line 2 (optional)
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            border={1}
            borderColor="black10"
            error={false}
            fontSize={3}
            inputRef={[Function]}
            mb={3}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            p={3}
            pb={2}
            placeholder="Add your apt, floor, suite, etc."
            returnKeyType="next"
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingBottom": 5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ]
            }
            value={undefined}
          />
        </View>
        <View
          mb={4}
          style={
            Array [
              Object {
                "marginBottom": 20,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={3}
            lineHeight={5}
            mb={2}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 5,
                },
                undefined,
              ]
            }
          >
            City
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            border={1}
            borderColor="black10"
            error={false}
            fontSize={3}
            inputRef={[Function]}
            mb={3}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            p={3}
            pb={2}
            placeholder="Add your city"
            returnKeyType="next"
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingBottom": 5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ]
            }
            value={undefined}
          />
        </View>
        <View
          mb={4}
          style={
            Array [
              Object {
                "marginBottom": 20,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={3}
            lineHeight={5}
            mb={2}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 5,
                },
                undefined,
              ]
            }
          >
            State, Province, or Region
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            border={1}
            borderColor="black10"
            error={false}
            fontSize={3}
            inputRef={[Function]}
            mb={3}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            p={3}
            pb={2}
            placeholder="Add state, province, or region"
            returnKeyType="next"
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingBottom": 5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ]
            }
            value={undefined}
          />
        </View>
        <View
          mb={4}
          style={
            Array [
              Object {
                "marginBottom": 20,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={3}
            lineHeight={5}
            mb={2}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 5,
                },
                undefined,
              ]
            }
          >
            Postal code
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            border={1}
            borderColor="black10"
            error={false}
            fontSize={3}
            inputRef={[Function]}
            mb={3}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            p={3}
            pb={2}
            placeholder="Add your postal code"
            returnKeyType="next"
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingBottom": 5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ]
            }
            value={undefined}
          />
        </View>
        <View
          mb={4}
          style={
            Array [
              Object {
                "marginBottom": 20,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={3}
            lineHeight={5}
            mb={2}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 5,
                },
                undefined,
              ]
            }
          >
            Country
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            border={1}
            borderColor="black10"
            hitSlop={undefined}
            mb={3}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            p={3}
            pb={2}
            style={
              Array [
                Object {
                  "borderColor": "#E5E5E5",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "marginBottom": 10,
                  "paddingBottom": 5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ]
            }
            testID={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              color="black30"
              ellipsizeMode="tail"
              fontSize={3}
              lineHeight={5}
              style={
                Array [
                  Object {
                    "color": "#C2C2C2",
                    "fontFamily": "AGaramondPro-Regular",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                ]
              }
            >
              Select your country
            </Text>
          </View>
        </View>
        <View
          height={50}
          style={
            Array [
              Object {
                "height": 50,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 1)",
                "borderRadius": 2,
                "flex": 1,
                "justifyContent": "center",
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={null}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                    Object {
                      "color": "white",
                      "opacity": 1,
                    },
                    Object {
                      "fontFamily": "Unica77LL-Medium",
                    },
                  ]
                }
              >
                Add billing address
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
